src/shelf_usage/zone.cc

```diff
@@ -554,6 +554,13 @@ void Zone::free(Offset block) {
     zoneheader->free_list[level].push(header_ptr, block/nvmm_read(&zoneheader->min_obj_size));
 }
 
+void Zone::clear_data(Offset block) {
```

Comment:
You need to remove memset from alloc call, which should be dependent on a flag passed on heap->Open().
Add a flag "NO_ZEROING_ON_ALLOC" (or a better name) to heap->Open(). Based on that remove zeroing from alloc and zero from free.

Reply:
Will remove memset from alloc. I missed it adding in git changes. But regarding flag usage, then this flag in nvmm have a dependency with number of delayed free threads in OpenFAM. Number of delayed free cant be zero if memset is delegated from alloc.  Need to take are this dependency.